packages/playwright/src/util.ts

```diff
@@ -91,6 +91,12 @@ export function createFileFiltersFromArguments(args: string[]): TestFileFilter[]
   });
 }
 
+export function createProjectMatcher(projectNames?: string[], projectGrep?: string[]): Matcher {
+  const names = projectNames || [];
+  const regexps = (projectGrep || []).map(grep => forceRegExp(grep));
+  return (project: string) => (!names.length && !regexps.length) || names.includes(project) || regexps.some(re => re.test(project));
```

Comment:
I'd think that both project name filter and project grep filter should match, not just one of them. Or maybe disallow that case in the CLI right away?